k3s agent "only https:// URLs are supported, invalid scheme"

**Version:**
<!-- Provide the output from `k3s -v` -->


**K3s arguments:**

`[core@node1 ~]$ curl -sfL https://get.k3s.io | K3S_URL=http://myserver:6443 K3S_TOKEN=mynodetoken sh -` (note the `http://` in K3S_URL)

**Describe the bug**
Install script succeeds, but k3s-agent fails to start with `Jun 27 15:05:20 node1 k3s[2748]: time="2020-06-27T15:05:20.405611746Z" level=error msg="only https:// URLs are supported, invalid scheme: http://127.0.0.1:40035"`

**To Reproduce**
Run `curl -sfL https://get.k3s.io | K3S_URL=http://myserver:6443 K3S_TOKEN=mynodetoken sh -`

**Expected behavior**
The install script should bail rather than propagating an invalid parameter

**Actual behavior**
Script exits successfully, but `k3s agent` fails to start with the given parameters.

**Additional context / logs**
Providing an HTTPS `K3S_URL` (such as literally https://myserver:6443) makes it past this check, which suggests we're setting up a local proxy or something to `K3S_URL` matching the same schema. We then error out on *that*, which is somewhat confusing.

@ShylajaDevadiga how are you testing?

```
python38) brandond@dev01:~$ curl -sfL https://get.k3s.io | K3S_TOKEN=secret K3S_URL=http://test:6443 sh -
[ERROR]  Only https:// URLs are supported for K3S_URL (have http://test:6443)
```

@brandond I am trying to join an agent to a master.  
```
curl -sfL https://get.k3s.io | INSTALL_K3S_COMMIT=04f57e5e1da489278d039478f095deb74bcc9b54 K3S_URL=http://IP:6443 K3S_TOKEN=secret sh -
[ERROR]  Only https:// URLs are supported for K3S_URL (have http://IP:6443)
```
When i actually join an agent
```
curl -sfL https://get.k3s.io | INSTALL_K3S_COMMIT=04f57e5e1da489278d039478f095deb74bcc9b54 K3S_URL=http://3.136.26.126:6443 K3S_TOKEN=secret sh -s - agent
[INFO]  Using commit 04f57e5e1da489278d039478f095deb74bcc9b54 as release
[INFO]  Downloading hash https://storage.googleapis.com/k3s-ci-builds/k3s-04f57e5e1da489278d039478f095deb74bcc9b54.sha256sum
[INFO]  Skipping binary downloaded, installed k3s matches hash
```
Logs:
```
Jul 23 19:18:31 ip-172-31-9-149 k3s[3983]: time="2020-07-23T19:18:31.845350270Z" level=error msg="only https:// URLs are supported, invalid scheme: http://127.0.0.1:32841"
Jul 23 19:18:33 ip-172-31-9-149 k3s[3983]: time="2020-07-23T19:18:33.845610443Z" level=error msg="only https:// URLs are supported, invalid scheme: http://127.0.0.1:32841"
```

--- a/install.sh
+++ b/install.sh
@@ -141,6 +141,19 @@ escape_dq() {
     printf '%s' "$@" | sed -e 's/"/\\"/g'
 }
 
+# --- ensures $K3S_URL is empty or begins with https://, exiting fatally otherwise ---
+verify_k3s_url() {
+    case "${K3S_URL}" in
+        "")
+            ;;
+        https://*)
+            ;;
+        *)
+            fatal "Only https:// URLs are supported for K3S_URL (have ${K3S_URL})"
+            ;;
+    esac
+}
+
 # --- define needed environment variables ---
 setup_env() {
     # --- use command args if passed or create default ---
@@ -153,14 +166,6 @@ setup_env() {
                 if [ -z "${K3S_TOKEN}" ] && [ -z "${K3S_CLUSTER_SECRET}" ]; then
                     fatal "Defaulted k3s exec command to 'agent' because K3S_URL is defined, but K3S_TOKEN or K3S_CLUSTER_SECRET is not defined."
                 fi
-
-                case "${K3S_URL}" in
-                    https://*)
-                        ;;
-                    *)
-                        fatal "Only https:// URLs are supported for K3S_URL (have ${K3S_URL})"
-                        ;;
-                esac
                 CMD_K3S=agent
             fi
         ;;
@@ -170,6 +175,9 @@ setup_env() {
             shift
         ;;
     esac
+
+    verify_k3s_url
+
     CMD_K3S_EXEC="${CMD_K3S}$(quote_indent "$@")"
 
     # --- use systemd name if defined or create default ---

